### DISCUSSION TITLE
Use a config file record similar as what is in Mock

### DISCUSSION THREAD

**Discussion by kozaka-tv** [Th0rstenf/Rocksniffer-to-Streamer.bot-extension#92] [Ideas]
I think it would be a good idea, to have a similar Config record (or even to move the one from Mock to Program) for all the config values.

- Init time, we just need to load all the values
- In a ToString, we could log out all the information

Then the Init code part could be more simple (without all the logs and co).

Later, if it will be implemented, #84 could be done with ease.

**Comment by Th0rstenf** [comment]
Please don't move things to backlog before they're agreed on. It's not even clear what exactly your goal is and what it's trying to achieve.

We will most likely not use a config file at all. We could however add an abstraction for getting/setting a global var that includes the log. This would serve the same purpose

**Comment by kozaka-tv** [comment]
This is more or less 2 main topic:
1) Use a record in Programm for all the configs and pass it/make accessible for all the Programm processes
2) Make the installment smoother and easier for the user with a config file

Point 1) is just a refactoring, as it adds no new logic, just the code would be more abstract and clean.
Point 2) is good, in case we introduce a new variable, or something, and the user need to change that manually in SB. If it is a file, the user do not need to bother with the sub-actions in SB. Or do not need to "merge" everything in case he imports a new version later. Because we would have only the read config sub action, and the Programm itself.

That's what is behind my idea, and not just logging. The getter/setter logic is kind of already done. With ToString I mean, we could with a  one liner logg out the config.

Idk...is it possible to put a text in SB for config as a sub-action? I do not remember that this would be possible. Like a Comment or something. Then we could avoid the local config file.

But this is maybe what in #94 also mentioned. I did not get everything what you ment in #94.
